roles/spark/tasks/main.yml

```diff
@@ -80,11 +87,19 @@
     - /usr/bin/pyspark
   when: spark_dirs.stat.exists is defined and not spark_dirs.stat.exists
 
+- stat: path="{{ spark_src_dir }}/spark-{{ spark_version }}-bin-without-hadoop.tgz"
+  register: spark_dirs_sci
+  tags: ["scispark"]
+
 - name: Remove SciSpark jar so it is re-compiled for the new Spark version
   file:
     state: absent
-    path: "{{ scispark_dir }}/SciSpark/target/scala-2.11/SciSpark.jar"
-  when: spark_dirs.stat.exists is defined and not spark_dirs.stat.exists
+    path: "{{ item }}"
+  with_items:
+    - "{{ scispark_dir }}/SciSpark/target/scala-2.11/SciSpark.jar"
```

Comment:
Is it always version 2.11? Can you use a wilcard instead of hardcoding the version number?

Reply:
I created a variable called scala_release (2.11) which is differs from scala_version (2.11.8).
The wildcard would make things complicated when copying, moving or linking jars.